taskvine/src/manager/vine_manager_put.c

```diff
@@ -348,15 +348,26 @@ Send a single input file of any type to the given worker, and record the perform
 
 static vine_result_code_t vine_manager_put_input_file(struct vine_manager *q, struct vine_worker_info *w, struct vine_task *t, struct vine_file *f)
 {
-
 	int64_t total_bytes = 0;
 	int64_t actual = 0;
 	vine_result_code_t result = VINE_SUCCESS; //return success unless something fails below
-
 	timestamp_t open_time = timestamp_get();
 
-	switch (f->type) {
+	// search for workers with our file
+	char *id, *new_source;
+	struct vine_worker_info *peer;
+	HASH_TABLE_ITERATE(q->worker_table, id, peer){
+		if(hash_table_lookup(peer->current_files, f->cached_name))
+		{
+			debug(D_VINE, "Signaling this file to be requested from worker: %s:%d", peer->transfer_addr, peer->transfer_port);
+			debug(D_VINE, "Original source: %s", f->source);
+			new_source = string_format("worker://%s:%d/%s", peer->transfer_addr, peer->transfer_port, f->cached_name);
+			f->source = new_source;
```

Comment:
Changing the file object on the fly probably isn't a good idea, since later uses of the file will want to know the original source.  Try making a new temporary `vine_file` object with the desired properties, and call `vine_manager_put_input_file` on that.